src/ssl.c

```diff
@@ -13231,18 +13244,6 @@ int wolfSSL_Cleanup(void)
 
 #ifdef OPENSSL_EXTRA
     wolfSSL_RAND_Cleanup();
-
-    if (wc_LockMutex(&rand_methods_mutex) != 0) {
-        WOLFSSL_MSG("Bad Lock Mutex rand methods");
-        return BAD_MUTEX_E;
-    }
-    rand_methods = NULL;
-    wc_UnLockMutex(&rand_methods_mutex);
-
-    if (wc_FreeMutex(&rand_methods_mutex) != 0)
```

Comment:
Why remove this?  The mutex needs to be freed or it will leak memory.  I also think it makes sense to set gRandMethods to NULL on cleanup, since the user is most likely cleaning up their RAND_METHODS struct as well when this is called.

Reply:
It is done in wolfSSL_RAND_Cleanup()